src/v2/Apps/Authentication/Utils/helpers.ts

```diff
@@ -143,9 +143,22 @@ export async function apiAuthWithRedirectUrl(
   response: Response,
   redirectPath: URL
 ): Promise<URL> {
-  const redirectUrl = sd.APP_URL + redirectPath.pathname + redirectPath.search
+  let redirectURL = redirectPath
+
+  const allowedHosts = sd.AUTH_REDIRECT_ALLOWED_HOSTS?.split(",") || []
+
+  if (!allowedHosts.includes(redirectPath.hostname)) {
+    // If it's not, redirect to the homepage
+    redirectURL = new URL("/", sd.APP_URL)
+  }
+
+  // const strippedRedirectURL =
+  //   redirectURL.origin + redirectURL.pathname + redirectURL.search
```

Comment:
I'm not sure why the current behavior re-creates the URL via concatenation vs. just using the provided `URL`. I removed it, and tests pass/behavior looks OK.

If we want to keep this concatenation behavior, I can uncomment this line (and pass that `strippedRedirectURL` through).

Reply:
@joeyAghion what do you think about this?